envs/arrow-env.yaml

```diff
@@ -27,10 +29,10 @@ packages:
   - feedstock : https://github.com/AnacondaRecipes/thrift-cpp-feedstock
     git_tag: 09e3838f8caaa3ab9728b111620422dd524459e1
 {% endif %}
-{% if build_type == 'cuda' %}
+{% if build_type == 'cuda' and not s390x %}
   - feedstock : cudatoolkit    #[cudatoolkit == "11.2"]
 {% endif %}
-  - feedstock : grpc-cpp
-  - feedstock : pyarrow
+  - feedstock : grpc-cpp       #[not s390x or build_type == 'cpu']
```

Comment:
This is `(NOT s 390x) OR (build_type = CPU)` , correct? I was wondering about the scope of the not (as it _could_ also be `NOT (s390 OR type=cpu)` )

Reply:
This means grpc-cpp is to be built for all platforms except s390x or if it is s390x, then build only for `cpu`.

Reply:
Stepping back a bit to understand why this is an issue now and not earlier.
Were we sending --build-type cpu as a command line argument earlier and not now ?
Shall we (1) set build_type in conda build config yaml file as below ?
```
build_type:
  - cpu
  - cuda # [not s390x]
```
OR
pass build-type command line option for s390x builds ?

Reply:
In the upcoming months, we will have tensorflow-env.yaml, tensorboard-env.yaml, etc enabled for s390x also and then those files also will require these conditions inside those files. My intention would be to avoid some of these these conditions in each -env.yaml file, if makes sense.

Reply:
We discussed this on slack and in our scrum and decided that we will revert these conditions for now. CI systems are anyway running s390x builds for build_type cpu. And in open-ce 1.6, we can tackle this by blocking cuda build type for s390x in builder code itself.